source/kernel/src/registry/mod.rs

```diff
@@ -229,6 +266,29 @@ pub enum RegistrationError {
     RegistryFull,
 }
 
+/// Errors returned by [`Registry::connect`] and
+/// [`Registry::connect_with_hello`].
+pub enum ConnectError<D: RegisteredDriver> {
+    /// No [`RegisteredDriver`] of this type was found!
+    NotFound,
+    /// The remote [`RegisteredDriver`] rejected the connection.
+    Rejected(D::ConnectError),
+    /// The remote [`RegisteredDriver`] has been registered, but the service
```

Comment:
I wonder if we could check this in the registry (awaiting channel closure?), tho we can't remove services anyway yet.